crypto3/libs/zk/include/nil/crypto3/zk/snark/systems/plonk/placeholder/lookup_argument.hpp

```diff
@@ -589,10 +624,7 @@ namespace nil {
                         std::unordered_map<typename FieldType::value_type, std::size_t> sorting_map;
                         for (std::size_t i = 0; i < reduced_value.size(); i++) {
                             for (std::size_t j = 0; j < usable_rows_amount; j++) {
-                                if(sorting_map.find(reduced_value[i][j]) != sorting_map.end())
-                                    sorting_map[reduced_value[i][j]]++;
-                                else
-                                    sorting_map[reduced_value[i][j]] = 1;
+                                sorting_map[reduced_value[i][j]] = 1;
```

Comment:
Are we sure the values in reduced_value don't repeat? Maybe we keep something like:

 if(sorting_map.find(reduced_value[i][j]) != sorting_map.end())
                                  throw std::invalid_argument("Values in sorting map must not repeat");
                              else
                                  sorting_map[reduced_value[i][j]] = 1;

Reply:
Values in `reduced_value` may repeat.
Initially `sorting_map` contains number of each occurencies in `lookup_input` + 1.

When we meet some `v` in `reduced_value` for the first time during sorting procedure we pusht one occurrence from `reduced_value` and all from `reduced_input` to sorted columns.

Next times we'll push only one value from `reduced_value` and nothing from `reduced_input`. That's why we set sorting_map[value] = 1 after first occurence.

So, `sorting_map`  containt information -- how many times `v` should be pushed to sorted columns next time